tests/pipeline/test_earnings.py

```diff
@@ -57,30 +56,33 @@ def setUpClass(cls):
 
         cls.earnings_dates = {
             # K1--K2--E1--E2.
-            A: to_series(
-                knowledge_dates=['2014-01-05', '2014-01-10'],
-                earning_dates=['2014-01-15', '2014-01-20'],
-            ),
+            A: pd.DataFrame({
```

Comment:
Yeah, but maybe instead of reverting, we can test that promotion logic separately from events/buyback auth in a common test for events (`test_events.py`).